### PR TITLE
Only emit click events for left mouse button

### LINKED ISSUE #4566: onPress is fired for all mouse buttons
### Environment
1. `react-native -v`:
react-native-cli: 2.0.1
react-native: 0.61.5
2. `react-native run-windows --info`:
System:
    OS: Windows 10 10.0.18363
    CPU: (12) x64 Intel(R) Xeon(R) W-2133 CPU @ 3.60GHz
    Memory: 15.87 GB / 31.73 GB
  Binaries:
    Node: 10.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.7.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: 4.9.4 - C:\Users\bhamlin\bin\watchman.EXE
  npmPackages:
    react: 16.9.0 => 16.9.0
    react-native: ^0.61.5 => 0.61.5
    react-native-windows: 0.61.0-beta.71 => 0.61.0-beta.71
  Installed UWP SDKs:
    10.0.17763.0
    10.0.18362.0

### Steps to Reproduce

1. Create a button with an onPress handler.
2. Click it with right click, middle click, M4 (back), M5(forward)

### Expected Behavior

I expected nothing to happen

### Actual Behavior

The button is clicked.

### Additional context

There full-fledged mouse support discussed in https://github.com/microsoft/react-native-windows/issues/2752, but in the short term can we just not fire onPress for these events? It causes user confusion and definitely does not feel like the right thing.

### PR DESCRIPTION
**Pull request by rectified95**
Closes #4566 
#2752 Discusses full mouse support. This PR is to mitigate the unexpected firing of identical onPress events for all mouse buttons.

Caveat - this approach results in the "Ended a touch event which was not counted in `trackedTouchCount`." warning being logged for all clicks other than the left mouse ones - open to suggestions of other ways to do this. 

I've tested that this doesn't break touch events (using my Surface laptop). Pen follows the same logic.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4878)

### CODE REVIEW

File: vnext/ReactUWP/Views/TouchEventHandler.cpp [hunk 1]
```diff
@@ -88,7 +88,11 @@ void TouchEventHandler::OnPointerPressed(
     UpdatePointersInViews(instance, args, tag, sourceElement);
 
     size_t pointerIndex = AddReactPointer(args, tag, sourceElement);
-    DispatchTouchEvent(TouchEventType::Start, pointerIndex);
+
+    // For now, we only want to send click events for the left mouse button.
+    if (m_pointers[pointerIndex].isLeftButton) {
```
Comment (kaiguo) [inline vnext/ReactUWP/Views/TouchEventHandler.cpp:93]: Will this block finger taps?

Comment (rectified95) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:93]: This says that finger taps also produce the PointerPressed event 
(https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.pointerpressed?view=winrt-19041#remarks)

Comment (kmelmon) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:93]: Sure PointerPressed will be raised, but this code is explicitly checking the pointer to see if the left button, which I believe will not be the case for a touch event.

You can test touch on a non-touch device with a little utility app called MouseHookApp.  It converts mouse into touch.
Go here:
\\kmelmondev\public\MouseHookApp.exe
Once launched all mouse is converted to touch.  Kill the app to go back to normal.

Can you test out your change with MouseHookApp?


---
In reply to: [426197483](https://github.com/microsoft/react-native-windows/pull/4878#discussion_r426197483) [](ancestors = 426197483)

Comment (rectified95) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:93]: @kmelmon I've run your app, and the resulting touch events are still interpreted as left mouse button clicks:
image

Comment (kmelmon) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:93]: Interesting!  I wonder if the mouse button information is being preserved.  What happens if you use the right mouse button with MouseHookApp?

Comment (rectified95) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:96]: @kmelmon 
image
Device type is 'Mouse' not touch in this case, but button info is preserved.
Same happens if I press the wheel (middle button)

Comment (kmelmon) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:96]: I think that means that MouseHookApp isn't truly converting to pointer, and Kai's concern seems valid.  Sorry about that! 

I think you'll also need to query the deviceType, ie
if deviceType == pointer => let it through
if deviceType == mouse => let it through only if isLeftButton is true

Comment (rectified95) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:96]: I've just tried RD from my Surface, and tapping elements gives the same result as a left mouse click using the MouseHook app - deviceType=Touch, isLeftButton=True

Comment (kmelmon) [reply vnext/ReactUWP/Views/TouchEventHandler.cpp:97]: Discussed offline.  With the help of Ross Nichols from the input team, we determined that all the input types we care about including pen will supply isLeftonButton == True.  This change looks good to me.

---
In reply to: [428833625](https://github.com/microsoft/react-native-windows/pull/4878#discussion_r428833625) [](ancestors = 428833625)

**Pull request merged**